microprofile/config/src/test/java/io/helidon/microprofile/config/ConfigBeanDescriptorTest.java

```diff
@@ -0,0 +1,106 @@
+ * limitations under the License.
+ */
+package io.helidon.microprofile.config;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import io.helidon.config.mp.MpConfigSources;
+import io.helidon.microprofile.config.testsubjects.TestConfiguredBean;
+
+import jakarta.enterprise.inject.spi.AnnotatedType;
+import jakarta.enterprise.inject.spi.InjectionPoint;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.inject.ConfigProperties;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Lower level testing for {@link io.helidon.microprofile.config.ConfigBeanDescriptor}.
+ */
+class ConfigBeanDescriptorTest {
+
+    private static MutableMpTest.MutableSource source;
+    private static Config config;
+    private static io.helidon.config.Config emptyConfig;
+
+    @BeforeAll
+    static void initClass() {
+        config = createTestConfiguredBeanConfig();
+
+        // we need to ensure empty config is initialized before running other tests,
+        // as this messes up the mapping service counter
+        emptyConfig = io.helidon.config.Config.empty();
+
+        source = new MutableMpTest.MutableSource("initial");
+
+        // register config
+        ConfigProviderResolver configProvider = ConfigProviderResolver.instance();
+        configProvider.registerConfig(config, Thread.currentThread().getContextClassLoader());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void testProduceListFields() throws Exception {
+        AnnotatedType<?> annotatedType = mock(AnnotatedType.class);
+        when(annotatedType.getJavaClass()).thenReturn((Class) TestConfiguredBean.class);
+
+        Field strListField = Objects.requireNonNull(TestConfiguredBean.class.getField("strList"));
+        Type strListType = strListField.getGenericType();
+        Field intListField = Objects.requireNonNull(TestConfiguredBean.class.getField("intList"));
+        Type intListType = intListField.getGenericType();
+
+        ConfigProperties configProperties = TestConfiguredBean.class.getAnnotation(ConfigProperties.class);
+        assertNotNull(configProperties);
+
+        ConfigBeanDescriptor descriptor = ConfigBeanDescriptor.create(annotatedType, configProperties);
+        assertNotNull(descriptor);
+
+        InjectionPoint strListInjectionPoint = mock(InjectionPoint.class);
+        when(strListInjectionPoint.getType()).thenReturn(strListType);
+
+        // the target of the test is here:
+        Object instance = descriptor.produce(strListInjectionPoint, config);
+        assertNotNull(instance);
```

Comment:
Please use Hamcrest (always) for assertions
Do static imports for Hamcrest methods.
Add assertion for instanceOf if it may produce a different type.

```
assertThat(o, instanceOf(TestConfiguredBean.class));
assertThat(o, notNullValue());
```